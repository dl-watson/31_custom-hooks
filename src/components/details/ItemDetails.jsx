import React from "react";
import PropTypes from "prop-types";
import titleCase from "../../utils/titleCase";

const ItemDetails = ({ character }) => {
  [character] = character;
  return (
    <div>
      {titleCase(character.pokemon)}
      <img src={character.url_image} alt={titleCase(character.pokemon)} />
      <div>
        <p>Height: {character.height}</p>
        <p>Weight: {character.weight}</p>
        <p>Primary Type: {character.type_1}</p>
        <p>Secondary Type: {character.type_2}</p>
        <p>Pokebase: {character.pokebase}</p>
      </div>
    </div>
  );
};

ItemDetails.propTypes = {
  character: PropTypes.array,
};

export default ItemDetails;
