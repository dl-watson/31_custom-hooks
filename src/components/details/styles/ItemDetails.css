* {
    margin: 0;
    padding: 0;
}

.ItemDetails {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    text-align: left;

    heading {
        font-size: 2rem;
    }

    img {
        height: 60vw;
    }

    section {
        margin: 20px;
    }
}