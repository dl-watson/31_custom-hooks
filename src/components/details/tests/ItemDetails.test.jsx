global.fetch = require("node-fetch");
import React from "react";
import { render, cleanup } from "@testing-library/react";
import ItemDetails from "../ItemDetails";

describe("ItemDetails component", () => {
  afterEach(() => cleanup());
  it("renders ItemDetails", () => {
    const { asFragment } = render(
      <ItemDetails character={[{ pokemon: "pokemon" }]} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
