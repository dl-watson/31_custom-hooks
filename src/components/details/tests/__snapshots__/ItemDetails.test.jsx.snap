// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDetails component renders ItemDetails 1`] = `
<DocumentFragment>
  <main
    class="ItemDetails"
    style="background-color: rgb(212, 208, 207); color: rgb(68, 75, 77);"
  >
    <article>
      Pokemon
    </article>
    <img
      alt="Pokemon"
    />
    <section
      class="details"
    >
      <p>
        Height: 
      </p>
      <p>
        Weight: 
      </p>
      <p>
        Primary Type: 
      </p>
      <p>
        Secondary Type: 
      </p>
      <p>
        Pokebase: 
      </p>
    </section>
  </main>
</DocumentFragment>
`;
